fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchIncludeTest.java

```diff
@@ -0,0 +1,585 @@
+        Response response =
+                target.path("Patient")
+                .queryParam("_tag", tag)
+                .queryParam("gender", "female")
+                .queryParam("_include", "Patient:link")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Bundle bundle = response.readEntity(Bundle.class);
+
+        assertNotNull(bundle);
+        assertEquals(2, bundle.getEntry().size());
+        assertEquals(patient2Id, bundle.getEntry().get(0).getResource().getId());
+        assertEquals(SearchEntryMode.MATCH, bundle.getEntry().get(0).getSearch().getMode());
+        assertEquals(patient1Id, bundle.getEntry().get(1).getResource().getId());
+        assertEquals(SearchEntryMode.INCLUDE, bundle.getEntry().get(1).getSearch().getMode());
+    }
+
+    @Test(groups = { "server-search-include" }, dependsOnMethods = {"testCreatePatient2"})
+    public void testSearchIncludeMultipleIncludedResults() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_tag", tag)
+                .queryParam("gender", "female")
+                .queryParam("_include", "Patient:organization")
+                .queryParam("_include", "Patient:link")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Bundle bundle = response.readEntity(Bundle.class);
+
+        assertNotNull(bundle);
+        assertEquals(3, bundle.getEntry().size());
+        assertEquals(patient2Id, bundle.getEntry().get(0).getResource().getId());
+        assertEquals(SearchEntryMode.MATCH, bundle.getEntry().get(0).getSearch().getMode());
+        List<String> resourceIds = new ArrayList<>();
+        for (int i=1; i<bundle.getEntry().size(); ++i) {
+            resourceIds.add(bundle.getEntry().get(i).getResource().getId());
+            assertEquals(SearchEntryMode.INCLUDE, bundle.getEntry().get(i).getSearch().getMode());
+        }
+        assertTrue(resourceIds.contains(patient1Id));
+        assertTrue(resourceIds.contains(organization1Id));
+    }
+
+    @Test(groups = { "server-search-include" }, dependsOnMethods = {"testCreatePatient2"})
+    public void testSearchIncludeWildcardMultipleIncludedResults() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_tag", tag)
+                .queryParam("gender", "female")
+                .queryParam("_include", "Patient:*")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Bundle bundle = response.readEntity(Bundle.class);
+
+        assertNotNull(bundle);
+        assertEquals(4, bundle.getEntry().size());
```

Comment:
would this fail after @AfterClass fails to execute with the removal of prior test failures?

Reply:
I won't comment on the rest of them, just food for thought

Reply:
The searches all get scoped to the set of resources created for a single test run by using a tag based on an Instant.now() value, so prior test artifacts should not be an issue, even if not cleaned up properly.

Reply:
excellent!